src/stores/WalletStore.ts

```diff
@@ -158,13 +155,27 @@ class WalletStore extends PersistableStore {
       void this.handleAccountChanged()
     })
     provider.on('disconnect', (error: unknown) => {
+      // Sometimes this error fires when user switching between networks too fast, we should not clean data in this case
+      if (
+        error instanceof Error &&
+        error.message.includes('Attempting to connect.')
+      ) {
+        handleError(error)
+        return
+      }
+
       if (provider) provider.removeAllListeners()
       handleError(error)
       this.clearData()
     })
     provider.on('chainChanged', async () => {
+      if (await this.isNetworkRight()) {
+        this.needNetworkChange = false
+        this.account = await this.getAccount()
+        return
+      }
       this.account = undefined
-      await this.connect()
+      this.needNetworkChange = true
```

Comment:
Looks like you're doing same thing in `.connect` method (I mean cleaning the `localstorage` data), let's extract it into function like `private resetData`

Reply:
wdym by "cleaning localstorage data"? I'm changing only "account" and "needNetworkChange" value